### DISCUSSION TITLE
Install vc_redist

### DISCUSSION THREAD

**Discussion by mtissington** [wixtoolset/issues#7641] [Questions]
There are many hits on stackoverflow about installing vc_redist with Wix 3.x but not 4.x

I'm trying to use an `ExePackage` to do the install but it's complaining thar I must specify `DetectCondition`
In the past launching the vc_redist handled itself.

So what do I need to specify for `DetectCondition` when install vc_redist?

Thanks.

**Comment by harper23** [comment]
I used an ExecPackage in WiX3.x, but never found a DetectCondition as necessary. You could show the actual error message and the source code that causes the error. That can help to reproduce your problem and gives more chance to get an answer. Try a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) .

Anyway, that's the excerpt from my bundle source code:

```
    <Chain>
      <ExePackage
        SourceFile="$(env.BUILD_DIR)\Microsoft\VC12\Redist\vcredist_x86.exe"
        InstallCommand="/q ACTION=Install"
        RepairCommand="/q ACTION=Repair /hideconsole" />
      <MsiPackage
        SourceFile="$(var.Installation.TargetDir)Application.msi" />
    </Chain>
```

**Comment by mtissington** [comment]
Yes, I did the same or similar - but in Wix 4 a value for DetectCondition is now required (the documentation is very clear about this too)

**Reply by robmen** [reply]
And it should have been required in v3. If you have an ExePackage element in v3, it really needs a DetectCondition attribute to work correctly.

**Reply by mtissington** [reply]
I was hoping you were going to. tell me what the condition is for VS redist :)

**Reply by robmen** [reply]
Isn't VC_redist a Bundle? If so, use the v4 BundlePackage. It does the work for you.

**Reply by mtissington** [reply]
Hmm, now I'm confused. - how do I know if it's a Bundle? and where can I find a list of Bundles (say for .NET too)?

**Comment by pas059** [comment]
Hi,
on my side i a use a bootstrapper project created under Visusal studio and the msi packages for the VC runtimes; my Bundle.wxs file contains the following:
```
<Chain>
....
<!-- MSVC v143 runtime x64 -->
<MsiPackage Id="MSVC_v143" SourceFile="C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Redist\MSVC\v143\vc_redist_msi\vc_redist.x64.msi" Compressed="yes" Vital="no" />
...
</Chain>
```
that's all
